aws-traffic-mirroring/templates/vpc-mirroring-target-instance.yaml

```diff
@@ -0,0 +1,151 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: Seekret EC2 Instance that acts as a VPC Traffic Mirroring Target
+
+Parameters:
+  LatestAmiId:
+    Type: 'AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>'
+    Default: '/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2'
+  CustomerVpcId:
+    Type: String
+    Description: The ID of the Customer's VPC
+  SourceVpcIpv4Cidr:
+    Type: String
+    AllowedPattern: "(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})/(\\d{1,2})"
+    Description: Customer VPC's Cidr
+    ConstraintDescription: 'must be a valid CIDR range of the form x.x.x.x/x.'
+  CustomerSubnetId:
+    Type: String
+    Description: The subnet Id of the Customer's ALB
+  BucketS3Url:
+    Type: String
+    Description: Bucket S3 URL
+    Default: https://s3.amazonaws.com
+  BucketName:
+    Type: String
+    Description: Bucket name
+  BucketAccessKey:
+    Type: String
+    Description: Access key for the bucket
+  BucketSecretKey:
+    Type: String
+    Description: Secret key for the bucket
+  BucketFolderName:
+    Type: String
+    Description: Folder name to store the pcaps in the bucket
+    Default: default/captures
+  RotationSeconds:
+    Type: String
+    Description: Interval to rotate files in the sniffer
+    Default: 600
+  MaxFileSize:
+    Type: String
+    Description: Max file size before creating a new pcap file
+    Default: 50
+
+Resources:
+  SecurityGroup:
+    Type: "AWS::EC2::SecurityGroup"
+    Properties:
+      GroupDescription: !Sub Security group for ${AWS::StackName} instance
+      GroupName: !Sub ${AWS::StackName}-sg
+      VpcId: !Ref CustomerVpcId
+      SecurityGroupIngress:
+        - { IpProtocol: udp, FromPort: 4789, ToPort: 4789, CidrIp: !Ref SourceVpcIpv4Cidr, Description: 'Allow mirrored traffic from VPC' }
+      Tags:
+        - { Key: Name, Value: !Sub '${AWS::StackName}-sg' }
+
+  Role:
+    Type: "AWS::IAM::Role"
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          -
+            Effect: Allow
+            Principal:
+              Service: ec2.amazonaws.com
+            Action: sts:AssumeRole
+      Path: /
+      RoleName: !Sub ${AWS::StackName}-role
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore
+
+  RoleProfile:
+    Type: AWS::IAM::InstanceProfile
+    Properties:
+      Path: /
+      Roles:
+        - !Ref Role
+      InstanceProfileName: !Sub '${AWS::StackName}-profile'
+
+  LaunchTemplate:
+    Type: "AWS::EC2::LaunchTemplate"
+    Metadata:
+      AWS::CloudFormation::Init:
+        config:
+          files:
+            /home/seekret-conf.env:
+              content: !Sub |
+                  ACCESS_KEY=${BucketAccessKey}
+                  SECRET_ACCESS_KEY=${BucketSecretKey}
+                  S3_URL=${BucketS3Url}
+                  BUCKET_NAME=${BucketName}
+                  FOLDER_NAME=${BucketFolderName}
+                  BPF_FILTER=tcp and port 4789
```

Comment:
what is port 4789?
why not setting the BPF filter as parameter as well?

Reply:
that's the port for aws traffic mirroring, changing it will prevent traffic mirroring from working..
if we make the change to use a dedicated network interface, this parameter can be omitted and we don't need any bpf filter